### PR TITLE
feat: add junit output support to scorecard-classic

### PR DESCRIPTION
**Pull request by RomanHotsiy**
## What/Why/How?

Adds `junit` as a supported `scorecard-classic` output format so CI systems can ingest scorecard results as test reports.

Implementation details:
- add a dedicated JUnit formatter for `scorecard-classic`
- wire `--format=junit` into the CLI command and argument typing
- cover the formatter with unit tests for empty, error, warning, escaping, and missing-location cases
- document the new output format and add a changeset

## Reference

N/A

## Testing

- `npx vitest run packages/cli/src/commands/scorecard-classic/__tests__/checkstyle-formatter.test.ts packages/cli/src/commands/scorecard-classic/__tests__/json-formatter.test.ts packages/cli/src/commands/scorecard-classic/__tests__/stylish-formatter.test.ts packages/cli/src/commands/scorecard-classic/__tests__/junit-formatter.test.ts`
- `npx tsc --noEmit --skipLibCheck`
- pre-commit hooks ran `npm run lint`, `npm run lint:fix`, and `npm run format`

## Screenshots (optional)

N/A

## Check yourself

- [x] This PR follows the [contributing guide](https://github.com/Redocly/redocly-cli/blob/main/CONTRIBUTING.md#pull-request-guidelines) 
- [x] All new/updated code is covered by tests
- [ ] Core code changed? - Tested with other Redocly products (internal contributions only)
- [ ] New package installed? - Tested in different environments (browser/node)
- [x] Documentation update has been considered 

## Security

- [x] The security impact of the change has been considered 
- [x] Code follows company security practices and guidelines

---

> [!NOTE]
> **Low Risk**
> Low risk additive change: introduces a new `junit` output formatter and wires it into `scorecard-classic` without altering validation logic. Main risk is output format correctness/escaping for CI consumers, mitigated by new unit tests.
> 
> **Overview**
> `scorecard-classic` now supports `--format=junit`, emitting JUnit XML with one suite per run and one case per issue (errors as failures, warnings as skipped), including escaped details and location fallbacks.
> 
> This wires the new formatter into CLI argument choices/types, factors ANSI-stripping into a shared `stripAnsiCodes` utility (reused by JSON output), adds dedicated unit tests for the JUnit output, and updates docs plus a changeset announcing the new format.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit de1255b2b36ef00c712144aaff69fa93e202537d. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by JLekawa — APPROVED** [review]

File: packages/cli/src/commands/scorecard-classic/types.ts [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
+export type ScorecardClassicOutputFormat =
+  | Extract<OutputFormat, 'stylish' | 'json' | 'checkstyle'>
+  | 'junit';
+
 export type ScorecardClassicArgv = {
   api: string;
   config: string;
   'project-url'?: string;
-  format: OutputFormat;
+  format: ScorecardClassicOutputFormat;
```
Comment (DmitryAnansky) [inline packages/cli/src/commands/scorecard-classic/types.ts:12]: Should we just extend `OutputFormat` with `junit` ?
```
export type OutputFormat =
  | 'codeframe'
  | 'stylish'
  | 'json'
  | 'checkstyle'
  | 'codeclimate'
  | 'summary'
  | 'github-actions'
  | 'junit'
  | 'markdown';
```

Or make new type cleaner  
```
export type ScorecardClassicOutputFormat =
  Extract<OutputFormat, 'stylish' | 'json' | 'checkstyle' | 'junit'>;
```

File: packages/cli/src/commands/scorecard-classic/formatters/junit-formatter.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    `<testsuites name="redocly scorecard-classic" tests="${problems.length}" failures="${failures}" errors="0" skipped="${skipped}">\n`
+  );
+  logger.output(
+    `<testsuite name="scorecard-classic" tests="${problems.length}" failures="${failures}" errors="0" skipped="${skipped}">\n`
+  );
+  logger.output('<properties>\n');
+  logger.output(`<property name="api" value="${xmlEscape(path)}" />\n`);
+  logger.output(`<property name="achievedLevel" value="${xmlEscape(achievedLevel)}" />\n`);
+  logger.output(
+    `<property name="targetLevelAchieved" value="${targetLevelAchieved ? 'true' : 'false'}" />\n`
```
Comment (AlbinaBlazhko17) [inline packages/cli/src/commands/scorecard-classic/formatters/junit-formatter.ts]: We don't print `targetLevelAchieved` property in formatted results. This property is used to check if passed `targetLevel` through args is achieved, if not we [log an error](https://github.com/Redocly/redocly-cli/blob/main/packages/cli/src/commands/scorecard-classic/index.ts#L151) and do not print in formatted results `achievedLevel`. See [the example in `json-formatter`](https://github.com/Redocly/redocly-cli/blob/main/packages/cli/src/commands/scorecard-classic/formatters/json-formatter.ts#L109).

File: packages/cli/src/commands/scorecard-classic/formatters/junit-formatter.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  );
+  logger.output('</properties>\n');
+
+  for (const problem of problems) {
+    const location = getProblemLocation(problem, path);
+    const level = problem.scorecardLevel || 'Unknown';
+    const message = stripAnsiCodes(problem.message);
+    const details = xmlEscape(formatProblemDetails(problem, location)).replaceAll('&#10;', '\n');
```
Comment (DmitryAnansky) [inline packages/cli/src/commands/scorecard-classic/formatters/junit-formatter.ts]: Will it make sense to escape each individual detail value and join afterwards, avoiding the encode-then-decode round-trip?
smth like inside the function:

```
...
    `Level: ${xmlEscape(problem.scorecardLevel || 'Unknown')}`,
    `Rule: ${xmlEscape(problem.ruleId)}`,
....
```

File: packages/cli/src/commands/scorecard-classic/types.ts (hunk 1 above)
Comment (RomanHotsiy) [reply packages/cli/src/commands/scorecard-classic/types.ts:12]: done

File: packages/cli/src/commands/scorecard-classic/formatters/junit-formatter.ts (hunk 2 above)
Comment (RomanHotsiy) [reply packages/cli/src/commands/scorecard-classic/formatters/junit-formatter.ts]: done, thanks

File: packages/cli/src/utils/strip-ansi-codes.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export function stripAnsiCodes(text: string): string {
```
Comment (AlbinaBlazhko17) [inline packages/cli/src/utils/strip-ansi-codes.ts:1]: I think, you can move this function to utils folder

File: packages/cli/src/commands/scorecard-classic/__tests__/json-formatter.test.ts [hunk 5]
```diff
@@ -188,7 +188,7 @@ describe('printScorecardResultsAsJson', () => {
   it('should strip ANSI codes from messages', () => {
     const problems: ScorecardProblem[] = [
       {
-        message: '\u001b[31mError message with color\u001b[0m',
+        message: '\u001b[1;31mError message with color\u001b[0m',
```
Comment (DmitryAnansky) [inline packages/cli/src/commands/scorecard-classic/__tests__/json-formatter.test.ts:191]: Was this change intentional?

File: packages/cli/src/commands/scorecard-classic/types.ts (hunk 1 above)
Comment (DmitryAnansky) [reply packages/cli/src/commands/scorecard-classic/types.ts:12]: I don't see the changes, maybe smth lost in the commit or have you decided to leave it as it is?

File: packages/cli/src/utils/strip-ansi-codes.ts (hunk 4 above)
Comment (RomanHotsiy) [reply packages/cli/src/utils/strip-ansi-codes.ts:1]: done

File: packages/cli/src/commands/scorecard-classic/__tests__/json-formatter.test.ts (hunk 5 above)
Comment (RomanHotsiy) [reply packages/cli/src/commands/scorecard-classic/__tests__/json-formatter.test.ts:191]: yes. To cover broader case (two escape codes, not one)

File: packages/cli/src/commands/scorecard-classic/types.ts (hunk 1 above)
Comment (RomanHotsiy) [reply packages/cli/src/commands/scorecard-classic/types.ts:12]: I made the new type cleaner as you asked: 

```
export type ScorecardClassicOutputFormat =
  Extract<OutputFormat, 'stylish' | 'json' | 'checkstyle' | 'junit'>;
```

**Review by DmitryAnansky — APPROVED** [review]
